### PR TITLE
Add metrics view for Prometheus

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: c2cwsgiutils/metrics.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        Should be defined in the specific provider
+        """
+        return []
+
+
+PROVIDERS_ = []
+
+
+POD_NAME = socket.gethostname()
+SERVICE_NAME = re.match("^(.+)-[0-9a-f]+-[0-9a-z]$", POD_NAME)
```
Comment (pvalsecc) [inline c2cwsgiutils/metrics.py]: I'd say you are missing a `+` before the `$`

File: c2cwsgiutils/metrics.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+def view_(request: pyramid.request.Request) -> pyramid.response.Response:
+    request.response.text = metrics()
+    return request.response
+
+
+NUMBER_RE = re.compile(r"^[0-9]+$")
+
+
+class MemoryMapProvider(Provider):
```
Comment (pvalsecc) [inline c2cwsgiutils/metrics.py]: Shouldn't you add this provider to PROVIDERS_ in the init function?

File: c2cwsgiutils/metrics.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        self.extend = extend
+
+    def get_data(self) -> List[Tuple[Dict[str, str], Union[int, float]]]:
+        """
+        Should be defined in the specific provider
+        """
+        return []
+
+
+PROVIDERS_ = []
```
Comment (pvalsecc) [inline c2cwsgiutils/metrics.py]: The underscore should be first for private stuff...

File: c2cwsgiutils/metrics.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    global PROVIDERS_
+    PROVIDERS_.append(provider)
+
+
+def init(config: pyramid.config.Configurator) -> None:
+    config.add_route("c2c_metrics", r"/metrics", request_method="GET")
+    config.add_view(view_, route_name="c2c_metrics", http_cache=0)
+
+
+def metrics() -> pyramid.response.Response:
```
Comment (pvalsecc) [inline c2cwsgiutils/metrics.py]: private function

File: c2cwsgiutils/metrics.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            result.append(
+                "{}{{{}}} {}".format(
+                    provider.name, ",".join(['{}="{}"'.format(*e) for e in attrib.items()]), value
+                )
+            )
+
+    return "\n".join(result)
+
+
+def view_(request: pyramid.request.Request) -> pyramid.response.Response:
```
Comment (pvalsecc) [inline c2cwsgiutils/metrics.py]: underscore first

File: c2cwsgiutils/metrics.py (hunk 1 above)
Comment (sbrunner) [reply c2cwsgiutils/metrics.py]: Arg, thanks

File: c2cwsgiutils/metrics.py (hunk 3 above)
Comment (sbrunner) [reply c2cwsgiutils/metrics.py]: Will be done

File: c2cwsgiutils/metrics.py (hunk 2 above)
Comment (sbrunner) [reply c2cwsgiutils/metrics.py]: Will be done

File: c2cwsgiutils/metrics.py (hunk 4 above)
Comment (sbrunner) [reply c2cwsgiutils/metrics.py]: Will be done

File: c2cwsgiutils/metrics.py (hunk 5 above)
Comment (sbrunner) [reply c2cwsgiutils/metrics.py]: Will be done

**Comment by sbrunner** [comment]
Fixed :-)

File: c2cwsgiutils/metrics.py (hunk 3 above)
Comment (pvalsecc) [reply c2cwsgiutils/metrics.py]: You forgot this one.

Comment (sbrunner) [reply c2cwsgiutils/metrics.py]: Oups, done :-)

**Pull request merged**